lib/driver.js

```diff
@@ -1118,8 +1118,22 @@ class XCUITestDriver extends BaseDriver {
 
     if (this.isRealDevice()) {
       await installToRealDevice(this.opts.device, this.opts.app, this.opts.bundleId, this.opts.noReset);
+      if (this.opts.otherApps) {
+        log.warn('Capability otherApps is only supported for Simulators');
+      }
     } else {
       await installToSimulator(this.opts.device, this.opts.app, this.opts.bundleId, this.opts.noReset);
+      if (this.opts.otherApps) {
+        let otherApps;
```

Comment:
Maybe we can parse it a bit earlier and turn it into a function? I feel like the code is branched out too much